File: 20190917/20190923/566. 重塑矩阵.cc
/*
 * Description: 重塑矩阵
 * Author: AskeyNil
 * CreateDate: 2019-09-23 09:53:54
 * LastEditors: AskeyNil
 *
 * *********************************
 * **                             **
 * **     　  你只有足够努力     　  **
 * **       才能看上去毫不费力       **
 * **                             **
 * *********************************
 *
 */

#include <iostream>
#include <vector>

using namespace std;

class Solution {
  public:
    vector<vector<int>> matrixReshape(vector<vector<int>> &nums, int r, int c) {
        int current_r = nums.size();
        if (current_r == 0)
            return nums;
        int current_c = nums[0].size();
        if (current_c == 0)
            return nums;
        if (current_c * current_r != r * c)
            return nums;
        vector<vector<int>> vv;
        for (size_t i = 0; i < r; i++) {
            vector<int> v;
            for (size_t j = 0; j < c; j++) {
                v.push_back(
                    nums[((i * c) + j) / current_c][((i * c) + j) % current_c]);
            }
            vv.push_back(v);
        }
        return vv;
    }
};

int main(int argc, char const *argv[]) {
    /* code */
    return 0;
}

File: 20190917/20190923/566. 重塑矩阵.py
'''
Description: 重塑矩阵
Author: AskeyNil
CreateDate: 2019-09-23 09:53:50
LastEditors: AskeyNil

*********************************
**                             **
**     　  你只有足够努力     　  **
**       才能看上去毫不费力       **
**                             **
*********************************

'''


class Solution:
    def matrixReshape(self, nums: [[int]], r: int, c: int) -> [[int]]:
        # 限制条件 特判
        current_r = len(nums)
        if current_r == 0:
            return nums
        current_c = len(nums[0])
        if current_c == 0:
            return nums
        if current_c * current_r != r * c:
            return nums
        return [[nums[((x*c)+y) // current_c][((x*c)+y) % current_c] for y in range(c)] for x in range(r)]


print(Solution().matrixReshape([[1, 2, 3, 4]], 2, 2))
